### ISSUE TITLE
Not clear how to break lines for elements to satisfy `rustfmt`

### ISSUE THREAD

**Issue by sanmai-NL** [lfairy/maud#75]
Sometimes HTML elements in templates exceed 100 characters. As far as I can tell, each element must be on a single line. This makes `rustfmt` sad.

**Comment by lfairy** [comment]
The parser used to ignore extra semicolons in templates, which would have solved the problem at hand. But I removed that feature because I thought the semicolons would confuse users who'd expect them to do something. It also makes the syntax less clean, which I find is a selling point of the library (especially compared to [Horrorshow], which works on stable but compromises on terseness and usability to do so). Both of those reasons still apply today, so I'm hesitant to add semicolons back to the syntax.

In other words, if you have any other ideas then I'm all ears.

[Horrorshow]: https://github.com/Stebalien/horrorshow-rs

**Comment by sanmai-NL** [comment]
What do you mean with semicolons having to mean something? They would separate statements, that is a meaning. Or do I misunderstand what you're trying to say?
Minimalistic syntax is great. But breaking the ability to wrap lines is a steep price, don't you think? I think this should be the overriding concern.

**Comment by sanmai-NL** [comment]
@lfairy: I think this issue should be be tagged ‘bug’ or ‘enhancement’ by the way. There's an issue to address.

**Comment by lfairy** [comment]
Ooh, now I get what you mean. I think we have have been talking past each other so far.

Earlier, I thought you meant that rustfmt messes up the formatting of templates, because it interprets `/` as a binary operator and so tries to squish everything into one line. But that can't be right -- when I run a simple test case through rustfmt, the code is left unchanged.

Now I'll try to answer what I think is your actual point. Maud has always supported wrapping lines. This works today, and pleases rustfmt:

```rust
html! {
    input type="text"
        data-foo="supercalifragilisticexpialidocious"
        data-bar="eventhoughthesoundofit"
        data-baz="issomethingquiteatrocious" /
}
```

This feature follows from the design of Rust's macro system. Since macros run on parsed tokens, not the original source code, they cannot know whether a line break was used or not. (Technically it's possible to guess this from spans/diagnostics but that's not a good idea.) It follows that none of Maud's syntax can be sensitive to whitespace.

Does this explanation answer your question? If so then I'll be happy to tag this as a documentation issue.

**Comment by sanmai-NL** [comment]
Thanks, that's clear. I did search the documentation for words like ‘line breaking’/‘wrapping’ etc. A note would be helpful, but it's not a shortcoming, as you explained.
